src/minlog.ts

```diff
@@ -115,7 +128,30 @@ export class MinLog {
     return maxLevelCode;
   }
 
-  async log(levelCodeOrName: MinLogLevel, ...args: MinLogArg[]): Promise<void> {
+  async flush(): Promise<void> {
+    await this._queueFlushing;
+
+    let deferred = _.deferred<void>();
+    this._queueFlushing = deferred.promise;
+
+    let flushed = false;
+    this._queue.push(async function() {
+      flushed = true;
+    });
+
+    // eslint-disable-next-line no-unmodified-loop-condition, @typescript-eslint/no-unnecessary-condition
+    while (!flushed) {
```

Comment:
Nice solution. This looks totally weird at first. Is this a common pattern?

Reply:
?